### DISCUSSION TITLE
Question regarding name server responses (current google .biz SERVFAIL problem)

### DISCUSSION THREAD

**Discussion by Onepamopa** [AdguardTeam/AdGuardHome#3207] [Q&A]
As you've probably noticed by now, google's DNS (8.8.8.8/8.8.4.4) has problems resolving .biz domains.

I had to turn on "Parallel requests" - and got a proper response from CF's name server (which is the 2nd in the list).
The questions is - what happens if google's DNS (or any other for that matter) answers first with SERVFAIL? Will the other DNS servers responses also be considered? If this is already handled properly - good, but if its not, it's something that should.

This is what happens at the moment trying to get a response from google on a .biz domain:
nslookup my.biz 8.8.8.8
Server:         8.8.8.8
Address:        8.8.8.8#53

** server can't find my.biz: SERVFAIL

nslookup my.biz 1.1.1.1
Server:         1.1.1.1
Address:        1.1.1.1#53

Non-authoritative answer:
Name:   my.biz
Address: 72.167.241.46

**Comment by ameshkov** [comment]
> The questions is - what happens if google's DNS (or any other for that matter) answers first with SERVFAIL? Will the other DNS servers responses also be considered? If this is already handled properly - good, but if its not, it's something that should.

Hm, you know, it's not handled and AGH will simply respond with what it gets first. I'd be grateful if you file a bug report with some information on how to reproduce this (we'll need some domain that always returns SERVFAIL, this is not the case with Google and .biz right now).

**Reply by Onepamopa** [reply]
I've got no idea how to find a domain that always returns a SERVFAIL, however, the response in AGH could probably be checked for that?
